lib/nmatrix/enumerate.rb

```diff
@@ -73,6 +73,8 @@ def map(&bl)
     return enum_for(:map) unless block_given?
     # NMatrix-jruby currently supports only doubles
     cp  = jruby? ? self : self.cast(dtype: :object)
+    cp.map!(&bl)
```

Comment:
Okay. Normally, it's good form to squash this kind of commit together with one that it reverts. Has anyone talked to you about how to use rebase before?

Normally, I'd expect all commits in a pull request to be rebased into a single commit. In this case, however, I think it's important to maintain the commit history. With that said, I think it could be cleaned up quite a bit to make it more legible for those looking back at it in the future and trying to understand your changes.